### DISCUSSION TITLE
Texture Compression

### DISCUSSION THREAD

**Discussion by CatCode79** [Davidster/ikari#30] [General]
Here are some resources about it:

ktx2 parsing library (used also by Bevy):
https://github.com/BVE-Reborn/ktx2

Introduction of the ASTC Texture Compression:
https://blog.en.uwa4d.com/2022/11/23/introduction-of-the-astc-texture-compression-format/

Texture Compression in 2020 (charts quality/speed):
https://aras-p.info/blog/2020/12/08/Texture-Compression-in-2020/

**Comment by CatCode79** [comment]
I discovered that, at least in the vulkan world, the BC texture compression is the most supported on desktop:
https://vulkan.gpuinfo.org/listfeaturescore10.php?platform=windows
(filters the search with the prefix textureCompression)

in wgpu there is the feature [TEXTURE_COMPRESSION_BC](https://docs.rs/wgpu/latest/wgpu/struct.Features.html#associatedconstant.TEXTURE_COMPRESSION_BC) in order to use it

**Reply by Davidster** [reply]
very nice! is this format well-supported on dx12?

**Reply by CatCode79** [reply]
I have not found any site similar to gpuinfo but for directx. But if you think about it, in principle they should be very similar: these statistics are based on the actual support of video cards for a certain feature, then it depends on the drivers (vulkan, directx or other) to tell us if this feature is supported and communicate correctly with the video card (and in last for us to implement the software part).
The support is probably not the same because the features table is not identical for each type of driver, but I think they are indicative anyway.

**Reply by Davidster** [reply]
yeah you're right, it's probably much more device-dependant than backend-dependant
